clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -175,8 +177,13 @@ protected abstract void onJoinComplete(int generation,
      */
     public void ensureCoordinatorReady() {
         while (coordinatorUnknown()) {
-            RequestFuture<Void> future = sendGroupCoordinatorRequest();
-            client.poll(future);
+            if (findCoordinatorFuture == null)
+                findCoordinatorFuture = sendGroupCoordinatorRequest();
+
+            client.poll(findCoordinatorFuture);
+
+            RequestFuture<Void> future = findCoordinatorFuture;
+            findCoordinatorFuture = null;
```

Comment:
The other case is a consumer wakeup before the GroupCoordinator request returns.